src/dbnode/encoding/istream.go

```diff
@@ -98,13 +104,23 @@ func (is *istream) ReadBits(numBits uint) (uint64, error) {
 		return 0, is.err
 	}
 	var res uint64
-	for numBits >= 8 {
-		byteRead, err := is.ReadByte()
+	numBytes := numBits / 8
+	numBits = numBits % 8
+	if numBytes > 0 {
+		// Use Read call rather than individual ReadByte calls since it has
+		// optimized path for when the iterator is aligned on a byte boundary.
+		bytes := is.buffer[0:numBytes]
+		read, err := is.Read(bytes)
 		if err != nil {
 			return 0, err
 		}
-		res = (res << 8) | uint64(byteRead)
-		numBits -= 8
+		if read == numBytesInInt64 {
+			res = binary.BigEndian.Uint64(bytes)
```

Comment:
Would endianess ever be a concern?

Reply:
It shouldn't because here we are just preserving the existing behavior - which is that this code that is representing 8 bytes as a `uint64` happens to be using big endian (i.e. most significant byte first). That's what the `res = (res << 8) | uint64(b)` is doing. 

See https://play.golang.org/p/Xz41AfnZVHS

Reply:
Any reason not to leave it as is?  Or is this faster?

Reply:
Wouldn't it be faster to inline the big endian call?
```go
b := bytes
res = uint64(b[7]) | uint64(b[6])<<8 | uint64(b[5])<<16 | uint64(b[4])<<24 |
		uint64(b[3])<<32 | uint64(b[2])<<40 | uint64(b[1])<<48 | uint64(b[0])<<56
```

Reply:
Yeah inlining is a good idea. And yeah I had added this because I suspected it could be faster and initial runtime tests suggested this was the case. I did some more rigorous time tests, though, and actually seems to be negligible so I'll change this to just use the loop no matter the size.